src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -866,17 +889,24 @@ protected virtual SecurityKey ResolveTokenDecryptionKey(string token, JsonWebTok
             if (validationParameters == null)
                 throw LogHelper.LogArgumentNullException(nameof(validationParameters));
 
+            StringComparison stringComparison = (validationParameters.TokenDecryptionKey is X509SecurityKey
+                || validationParameters.TokenDecryptionKey is ECDsaSecurityKey)
+                ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
             if (!string.IsNullOrEmpty(jwtToken.Kid))
```

Comment:
This is repeated and I got lost easily (there are two versions), suggest replacing with calls like the following:

        private static StringComparison GetCaseRuleIfX509(SecurityKey key) => key is X509SecurityKey ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
        private static StringComparison GetCaseRuleIfECDsa(SecurityKey key) => (key is X509SecurityKey || key is ECDsaSecurityKey) ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;

Reply:
Adding it in [1595e40](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1834/commits/1595e4097fc1b5c0f5b7a00fccf7f878bdd9c202)